### PR TITLE
Chat App

### LINKED ISSUE #8: Group Chat
When following streamers, be able
to chat with other common followers of the user (similar to the way twitch allows people to chat)

### PR DESCRIPTION
**Pull request by tarelmadar**
Chat App-
This PR demonstrates streamer's chat channel.
First, the user types the chat room name(streamer's chat page) he would like to enter.
The user transfers to a specific room, where he can talk with other users and the streamer itself.

More details about how everything works-
I specified a routing configuration specifically for WebSockets.
The routing listens on the specific path written in there and then handled by the chat consumer.
The chat consumer asynchronously handles all the different events written in there (which inherited from AsyncWebsocketConsumer).

### CODE REVIEW

**Comment by ifireball** [comment]
Since a previous PR I've seen about chat was much bigger I suspect this is not the entire chat app but a small part of it - but I'm aftraid the PR titme and description do not help me to understand what is included in this PR exactly. - also at 12 files and 163 lines - it is a little big.

File: sellotape/sellotape_dj/chat/templates/chat/room.html [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            const messageInputDom = document.querySelector('#chat-message-input');
+            const message = messageInputDom.value;
+            chatSocket.send(JSON.stringify({
+                'message': message
+            }));
+            messageInputDom.value = '';
+        };
+    </script>
+</body>
+</html>
```
Comment (lmilbaum) [inline sellotape/sellotape_dj/chat/templates/chat/room.html]: newline is missing

File: sellotape/sellotape_dj/chat/urls.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+from django.urls import path, re_path
+from .views import index, room
+
+
+app_name = 'chat'
+
+urlpatterns = [
+    path('', index, name = 'index'),
+    re_path(r'^(?P<room_name>[^/]+)/$', room, name='room'),
+]
```
Comment (lmilbaum) [inline sellotape/sellotape_dj/chat/urls.py]: newline

File: sellotape/sellotape_dj/chat/views.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+import json
+
+def index(request):
+    return render(request, 'chat/index.html')
+
+def room(request, room_name):
+    print(room_name)
+    return render(request, 'chat/room.html', {
+        'room_name': room_name
+    })
```
Comment (lmilbaum) [inline sellotape/sellotape_dj/chat/views.py]: newline

**Review by lmilbaum — CHANGES_REQUESTED** [review]
link the issue as I've suggested over Slack today

File: sellotape/sellotape_dj/chat/consumers.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        )
+
+    # Receive message from room group
+    async def chat_message(self, event):
+        message = event['message']
+
+        # Send message to WebSocket
+        await self.send(text_data=json.dumps({
+            'message': message
+        }))
```
Comment (lmilbaum) [inline sellotape/sellotape_dj/chat/consumers.py]: newline :-)

File: sellotape/sellotape_dj/chat/views.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
```
Comment (lmilbaum) [inline sellotape/sellotape_dj/chat/views.py:12]: something is wrong with the newline

Comment (lmilbaum) [reply sellotape/sellotape_dj/chat/views.py:12]: You might want to run flake8 on all files

File: sellotape/sellotape_dj/chat/templates/chat/index.html [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+            }
+        };
+
+        document.querySelector('#room-name-submit').onclick = function(e) {
+            var roomName = document.querySelector('#room-name-input').value;
+            window.location.pathname = '/chat/' + roomName + '/';
+        };
+    </script>
+</body>
+</html>
```
Comment (lmilbaum) [inline sellotape/sellotape_dj/chat/templates/chat/index.html]: newline

File: sellotape/sellotape_dj/chat/routing.py [hunk 7]
```diff
@@ -0,0 +1,7 @@
+from django.urls import re_path
+
+from . import consumers
+
+websocket_urlpatterns = [
+    re_path(r'ws/chat/(?P<room_name>\w+)/$', consumers.ChatConsumer),
+]
```
Comment (lmilbaum) [inline sellotape/sellotape_dj/chat/routing.py]: newline

**Review by lmilbaum — CHANGES_REQUESTED** [review]

**Review by tarelmadar — COMMENTED** [review]
fixes using flake8

**Review by lmilbaum — CHANGES_REQUESTED** [review]
Please link an issue to the PR

File: sellotape/sellotape_dj/chat/views.py (hunk 5 above)
Comment (lmilbaum) [reply sellotape/sellotape_dj/chat/views.py:12]: ??

**Pull request closed without merging**